src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -148,7 +153,8 @@ public void deleteRoles(final Node node) throws RepositoryException {
         authorizations.stream()
             .filter(x -> checkAccessTo.test(x) || checkAccessToClass.test(x))
             .forEach(x -> {
-                x.getAgents().stream()
+                Stream.concat(x.getAgents().stream(), dereferenceAgentClasses(x.getAgentClasses()).stream())
```

Comment:
I may be confused, but you are doing the checkAccessToClass above this forEach. Would it still be required? Or is that to allow for string agentClasses?